src/libslic3r/GCode.cpp

```diff
@@ -1728,7 +1728,11 @@ void GCode::_do_export(Print& print, GCodeOutputStream &file, ThumbnailsGenerato
     //m_placeholder_parser.set("has_single_extruder_multi_material_priming", has_wipe_tower && print.config().single_extruder_multi_material_priming);
     m_placeholder_parser.set("total_toolchanges", std::max(0, print.wipe_tower_data().number_of_toolchanges)); // Check for negative toolchanges (single extruder mode) and set to 0 (no tool change).
 
-    std::vector<unsigned char> is_extruder_used(print.config().filament_diameter.size(), 0);
+    // PlaceholderParser currently substitues non-existent vector values with the zero'th value, which is harmful in the
+    // case of "is_extruder_used[]" as Slicer may lie about availability of such non-existent extruder. We rather
+    // sacrifice 256B of memory before we change the behavior of the PlaceholderParser, which should really only fill in
+    // the non-existent vector elements for filament parameters.
+    std::vector<unsigned char> is_extruder_used(std::max(size_t(255), print.config().filament_diameter.size()), 0);
```

Comment:
Am I understanding correctly that the purpose of this change is to handle cases where the user tries to access a non-existent extruder in his custom G-code? For example, using `is_extruder_used[18]` when only 3 extruders/filaments are set in the slicer.

Reply:
Yes, I use it with my IDEX device (Snapmaker J1), and when only one filament is added in the slicer, it used to return true when attempting to check `is_extruder_used[1]` in the Start Gcode. Now it will return false.